all_seaing_utility/all_seaing_utility/rosbag_to_image.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+
+# python3 /home/arcturus/arcturus/dev_ws/src/all_seaing_vehicle/all_seaing_utility/all_seaing_utility/rosbag_to_image.py /home/arcturus/arcturus/test_05-21-24/test1/test1_0.db3 /zed/zed_node/rgb/image_rect_color /home/arcturus/arcturus/bag_output
+
+from argparse import ArgumentParser
+import cv2
+from cv_bridge import CvBridge
+import os
+import rosbag2_py
+from rclpy.serialization import deserialize_message
+from sensor_msgs.msg import Image
+import rclpy
+
+def main():
+    rclpy.init()
+
+    # so you can input info everytime, totally could hardcode tho
+    parser = ArgumentParser(description="Extract images from ROS2 bag")
+    parser.add_argument("bag_file", help="Input ROS2 bag file")
+    parser.add_argument("image_topic", help="Image topic to convert")
+    parser.add_argument("output_dir", help="Output directory")
+
+    args = parser.parse_args()
+
+    bag_file = args.bag_file
+    image_topic = args.image_topic
+    output_dir = args.output_dir
+
+    reader = rosbag2_py.SequentialReader()
+
+    storage_options = rosbag2_py.StorageOptions(uri=bag_file, storage_id='sqlite3')
+    converter_options = rosbag2_py.ConverterOptions(input_serialization_format='cdr', output_serialization_format='cdr')
+
+    reader.open(storage_options, converter_options)
+
+    bridge = CvBridge()
+    count = 0
+
+    print(f"Extracting images from {bag_file} on topic {image_topic} into {output_dir}")
+
+    choose_framerate = 5
+    current_rate = 579//38.314721344 #15
```

Comment:
Wondering how you got those numbers (was it from ros2 bag info?). I wonder if there is a way to calculate this value without hard-coding. If not, I'm also okay with a parameter like "sample_every" (e.g. sample_every 2 can get every 2 messages, just like how you are getting every framerate messages right now) and add it to the "parser.add_argument" above. I think ideally the people who use this script don't need to edit this file ever, just needs to provide the arguments.

Reply:
I got the numbers from ros2 bag info. In the new commit, I added the argument for fps and it finds fps through the code instead.